src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java

```diff
@@ -303,6 +306,46 @@ public static JsonArrayBuilder json(List<DataverseContact> dataverseContacts) {
         }
         return jsonArrayOfContacts;
     }
+    
+    public static JsonArrayBuilder getOwnersFromDvObject(DvObject dvObject) {
+        
+        List <DvObject> ownerList = new ArrayList();
+        dvObject = dvObject.getOwner(); // We're going to ignore the object itself
+        while (dvObject != null) {
+            ownerList.add(dvObject);
+            dvObject = dvObject.getOwner();
+        } 
+        
+        JsonArrayBuilder jsonArrayOfOwners = Json.createArrayBuilder();
+        
+        for (DvObject dvo : ownerList){
+                JsonObjectBuilder ownerObject = jsonObjectBuilder();
+                if (dvo.isInstanceofDataverse()){
```

Comment:
I would extract the type check to a separate method to improve readability, and try to use a `switch` statement instead of these `ifs`. As far as I know a `dvObject` can't be of more than one type, and by having these `ifs` without `else ifs` it seems that a dvObject can have multiple types which is confusing.

Reply:
I simplified it here because since we're ignoring the object itself we can only have a Dataverse or a Dataset